ducktape/tests/test.py

```diff
@@ -260,6 +273,7 @@ def __init__(self, **kwargs):
         self.log_collect = {}
 
         self._logger = None
+        self._scratch_dir = None
```

Comment:
What's the context for this change? It doesn't seem related to this patch at all and it is only used by tests. Why is this on the test rather than on services? Do the tests often have local temp data they need to work with?